controllers/admins.js

```diff
@@ -1,100 +1,82 @@
-  }
+  Admins.find({ name: req.params.email })
+    .then((matchedAdmins) => res.json({ matchedAdmins }))
+    .catch((error) => res.status(404).json(error));
 };
 
-const createNewAdmin = (req, res) => {
-  if (!req.query.email) {
-    res.status(400).send('Missing some data fields');
-  } else {
-    const newAdmin = {
-      id: (admins.length) + 1,
-      email: req.query.email,
-    };
-    admins.push(newAdmin);
-    fs.writeFile('./data/admins.json', JSON.stringify(admins), (err) => {
-      if (err) {
-        res.status(400).send('Couldn´t update data file');
-      } else {
-        res.status(201).send('New Admin succesfully created');
-      }
-    });
+const createAdmin = (req, res) => {
+  if (!req.body.email) {
+    return res.status(400).send('Please include the email to create a new Admin');
   }
-};
-
-const editAdminById = (req, res) => {
-  let adminPosition = -1;
-  let foundedAdmin = admins.find((admin, index) => {
-    if (admin.id === parseInt(req.params.id)) {
-      adminPosition = index;
-      return admin;
+  const newAdmin = new Admins({
+    email: req.body.email,
+  });
+  newAdmin.save((error, newAdmin) => {
+    if (error) {
+      return res.status(500).json({ msg: 'Error while creating a new Admin', error });
     }
+    return res.status(201).json(newAdmin);
   });
-  if (!foundedAdmin) {
-    res.status(400).send('This Id doesn´t exist');
+};
+
+const updateAdmin = (req, res) => {
+  if (!req.params.id) {
+    return res.status(400).json({ msg: 'You should enter a ID to be updated' });
   } else {
-    foundedAdmin = {
-      id: foundedAdmin.id,
-      email: req.query.email || foundedAdmin.email,
-    };
-    admins[adminPosition] = foundedAdmin;
-    fs.writeFile('./data/admins.json', JSON.stringify(admins), (err) => {
-      if (err) {
-        res.status(400).send('Couldn´t update data file');
-      } else {
-        res.status(200).send(`Admin with id ${req.params.id} succesfully edited`);
-      }
-    });
+    Admins.findByIdAndUpdate(
+      req.params.id,
+      {
+        email: req.body.email,
+      },
+      { new: true },
+      (error, updatedAdmin) => {
+        if (!updatedAdmin) {
+          return res.status(404).json({ msg: `Admin with the id: ${req.params.id} was not found` });
+        }
+        if (error) {
+          return res.status(400).json(error);
+        }
+        return res.status(200).json({ msg: 'Admin updated!', updatedAdmin });
+      },
+    );
   }
 };
 
-const deleteAdminById = (req, res) => {
-  let adminPosition = -1;
-  const foundedAdmin = admins.find((admin, index) => {
-    if (admin.id === parseInt(req.params.id)) {
-      adminPosition = index;
-      return admin;
-    }
-  });
-  if (!foundedAdmin) {
-    res.status(404).send(`There is no admin with this id: ${req.params.id}`);
+const deleteAdmin = (req, res) => {
+  if (!req.params.id) {
+    return res.status(400).json({ msg: 'You should enter an Admin ID to be deleted' });
```

Comment:
You could replace the `delete` method and use it like this instead adding multiple `else` or `else if`
```suggestion
const deleteAdmin = (req, res) => {
  if (!req.params.id) return res.status(400).json({ msg: 'You should enter an Admin ID to be deleted' });
  return Admins.findByIdAndDelete(req.params.id, (error, result) => {
    if (error) return res.status(400).json(error);
    if (result === null) return res.status(404).json({ msg: 'Admin ID not found' });
    return res.status(200).json({ msg: `Admin with the id: ${req.params.id} was deleted!` });
  });
};
```

Reply:
Done! It's very much readable now. Applied on every method with "if and else" statements. 👍